src/internal/operations/backup.go

```diff
@@ -338,6 +349,44 @@ func (op *BackupOperation) do(
 	return deets, nil
 }
 
+func Precheck(
+	ctx context.Context,
+	acc account.Account,
+	service path.ServiceType,
+	userID string,
+) error {
+	if service == path.SharePointService {
+		// No "enabled" check required for sharepoint
+		return nil
+	}
+
+	cred, err := acc.M365Config()
+	if err != nil {
+		return clues.Wrap(err, "getting creds")
+	}
+
+	client, err := api.NewClient(cred)
+	if err != nil {
+		return clues.Wrap(err, "constructing api client")
+	}
+
+	ui, err := client.Users().GetInfo(ctx, userID)
+	if err != nil {
+		return clues.Wrap(err, "unable to get user info")
+	}
+
+	if ui == nil || len(ui.ServicesEnabled) == 0 {
+		return graph.ErrServiceNotEnabled
+	}
+
+	_, ok := ui.ServicesEnabled[service]
+	if !ok {
+		return graph.ErrServiceNotEnabled
+	}
```

Comment:
I'd rather move this ownership out of operations and into GC.  After all, the clients being produced here already exist in the connector instance.

To keep in line with the other operations interfaces, I suggest extending the `BackupProducer` interface in `operations/inject.go` with a func like `IsBackupRunnable(ctx, service, resourceOwner) (bool, error)`. That way you can pass GC into this func directly as the interface supplier (or call IsBackupRunnable on line 149 and skip this Precheck func altogether).

Finally, adding testing around the IsBackupRunnable implementation in GC would be valuable.